go/vt/tabletserver/gorpcqueryservice/sqlquery.go

```diff
@@ -46,7 +46,9 @@ func (sq *SqlQuery) Rollback(ctx context.Context, session *proto.Session, noOutp
 // Execute is exposing tabletserver.SqlQuery.Execute
 func (sq *SqlQuery) Execute(ctx context.Context, query *proto.Query, reply *mproto.QueryResult) (err error) {
 	defer sq.server.HandlePanic(&err)
-	return sq.server.Execute(callinfo.RPCWrapCallInfo(ctx), query, reply)
+	execErr := sq.server.Execute(callinfo.RPCWrapCallInfo(ctx), query, reply)
+	tabletserver.AddTabletErrorToQueryResult(execErr, reply)
+	return nil
```

Comment:
You can't submit this: (new server / old client) combo won't work: the new server will populate Err, the old client will not look at it.